test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -529,13 +532,28 @@ public void testToQuery() throws IOException {
         }
     }
 
+    // TODO: Deprecate this method?
```

Comment:
I kept the `rewriteQuery` with the old signature because a bunch of downstream tests use it directly. Should we deprecate it though? It doesn't really fully simulate how queries are rewritten in production.